manager/state/raft/raft.go

```diff
@@ -115,6 +122,8 @@ type Node struct {
 	stopMu sync.RWMutex
 	// used for membership management checks
 	membershipLock sync.Mutex
+	// used to set Node
+	nodeLock sync.Mutex
```

Comment:
@abronan: The race detector doesn't seem to complain for me when I remove this lock. But I did two things to make `n.Node` safe to use without a lock:

- Set `isMember` only after `RestartNode` is called (`RestartNode` creates a goroutine, so this is correct and won't cause a deadlock).
- Always check `IsMember` before `IsStopped`.